test/e2e_node/criproxy_test.go

```diff
@@ -0,0 +1,77 @@
+//go:build linux
+// +build linux
+
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2enode
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/onsi/ginkgo/v2"
+	"github.com/onsi/gomega"
+
+	runtimeapi "k8s.io/cri-api/pkg/apis/runtime/v1"
+	"k8s.io/kubernetes/test/e2e/framework"
+	"k8s.io/kubernetes/test/e2e_node/criproxy"
+	admissionapi "k8s.io/pod-security-admission/api"
+)
+
+// Examples of using CRI proxy
+var _ = SIGDescribe("CRI Proxy", framework.WithSerial(), func() {
+	f := framework.NewDefaultFramework("image-manager")
+	f.NamespacePodSecurityLevel = admissionapi.LevelPrivileged
+
+	ginkgo.AfterEach(func() {
+		resetCRIProxyInjector()
+	})
+
+	ginkgo.It("should pull images failed", func(ctx context.Context) {
+		_, is, err := getCRIClient()
+		framework.ExpectNoError(err)
+		expectedErr := fmt.Errorf("PullImage failed")
+		addCRIProxyInjector(func(apiName string) error {
+			if apiName == criproxy.PullImage {
+				return expectedErr
+			}
+			return nil
+		})
+
+		_, err = is.PullImage(context.Background(), &runtimeapi.ImageSpec{Image: agnhostImage}, nil, nil)
```

Comment:
same comment as above. I would expect this test to create a Pod and check that it recieved some image pull error failure.